cpp/src/arrow/compute/api_vector.cc

```diff
@@ -325,6 +344,22 @@ Result<std::shared_ptr<Array>> DropNull(const Array& values, ExecContext* ctx) {
   return out.make_array();
 }
 
+Result<std::shared_ptr<Array>> CumulativeSum(const Array& values,
+                                             const CumulativeSumOptions& options,
+                                             ExecContext* ctx) {
+  ARROW_ASSIGN_OR_RAISE(Datum out,
+                        CallFunction("cumulative_sum", {Datum(values)}, &options, ctx));
+  return out.make_array();
+}
+
+Result<std::shared_ptr<Array>> CumulativeSum(const ChunkedArray& chunked_array,
+                                             const CumulativeSumOptions& options,
+                                             ExecContext* ctx) {
+  ARROW_ASSIGN_OR_RAISE(
+      Datum out, CallFunction("cumulative_sum", {Datum(chunked_array)}, &options, ctx));
+  return out.make_array();
```

Comment:
Need to collapse these functions into a [single one that takes in Datum as declared in the header file](https://github.com/apache/arrow/pull/12460/files#diff-d8e3089c085273e425a5106e77566f2e1348c483a79eeb544a4c9ee0e8327f34R544).